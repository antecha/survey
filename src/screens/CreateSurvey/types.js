// @flow
export type { Element } from 'react';

export type SurveyState = {
  showAddQuestion: boolean,
  saveQuestion: string,
};
