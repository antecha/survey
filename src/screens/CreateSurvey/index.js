// @flow
import React, { Component } from 'react';
import { Tabs, Input, Button } from 'antd';
import type { Element } from '../types';
import type { SurveyState } from '../types';

const { TabPane } = Tabs;

class CreateSurvey extends Component<{}, SurveyState> {
  state = {
    showAddQuestion: false,
    saveQuestion: '',
  };

  handleShowQuestion4444 = () => {
    this.setState({ showAddQuestion: true });
  };

  handleQuestion = e => {
    console.log(e);
    this.setState({ saveQuestion: e.target.value });
  };

  render(): Element<'div'> {
    const { showAddQuestion, saveQuestion } = this.state;

    console.log(showAddQuestion, saveQuestion);
    return (
      <div>
        <Tabs defaultActiveKey="1">
          <TabPane tab="Create Survey" key="1">
            {showAddQuestion ? (
              <Input onChange={this.handleQuestion} />
            ) : (
              <Button onClick={this.handleShowQuestion4444}>add question</Button>
            )}
          </TabPane>
          <TabPane tab="Preview Survey" key="2">
            {saveQuestion}
          </TabPane>
        </Tabs>
      </div>
    );
  }
}

export default CreateSurvey;
