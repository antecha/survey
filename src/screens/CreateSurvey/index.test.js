import React from 'react';
import { create } from 'react-test-renderer';
import CreateSurvey from './index';

jest.mock('antd', _ => ({
  ...require.requireActual('antd'),
  Tabs: () => <div />,
}));

describe('CreateSurvey', () => {
  const testRenderer = create(<CreateSurvey />);

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
