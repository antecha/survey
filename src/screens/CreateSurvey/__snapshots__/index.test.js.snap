// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateSurvey renders 1`] = `
<div>
  <div />
</div>
`;
